ClimbingGearBackend/Infrastructure/EFGearRepository.cs

```diff
@@ -0,0 +1,53 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using ClimbingGearBackend.Models;
+using ClimbingGearBackend.Interfaces;
+
+namespace ClimbingGearBackend.Infrastucture
+{
+  public class EFGearRepository : IGearRepository
+  {
+    private readonly ClimbingGearContext _dbContext;
+
+    public EFGearRepository(ClimbingGearContext dbContext)
+    {
+      _dbContext = dbContext;
+    }
+
+    public bool Exists(long id)
+    {
+      return _dbContext.Gear.Any(e => e.Id == id);
```

Comment:
Is there a way to directly query by ID instead of these Linq methods which take a lambda? Something where you can pass in the ID directly? Same thing applies to GetByIdAsync below.

I question the efficiency of this method, but I don't know the library you're using.

Reply:
Entity generally wants you to use lamda's so it can enforce type checking. There is a `findAsync` method on the `DbSet` class but I want it to return a bool. Whether either is faster is hard to say. I'm pretty sure they both craft a SQL statement to query the database with unless the value is in cache.

Reply:
I would double check that because it would seem magical to me that it can convert a lambda (a C# construct) into SQL (a text construct).

Reply:
So every LINQ statement seems to send a new query to the database. It even sets up caching and change tracking for each one usually. It is not scanning it's local context for whether that value exists.
https://docs.microsoft.com/en-us/ef/core/querying/how-query-works